users/views.py

```diff
@@ -1,3 +1,139 @@
+        password = request.POST.get("password")
+
+        try:
+            user = User.objects.get(email=email)
+        except Exception:
+            messages.error(request, "user does not exist")
+
+        user = authenticate(request, email=email, password=password)
+
+        if user is not None:
+            login(request, user)
+
+            return redirect("home")
+
+        else:
+            messages.error(request, "username or password does not exists")
+
+    form = MyUserCreationForm()
+
+    return render(request, "login/register_login.html", {'form': form, 'page': 'login'})
+
+
+def register_page(request):
+    """ return the login-register page with the register tab opened.
+
+        the user will have to input: username, email, password, confremation password,
+        and optionaly a photo for there profile and a phone number.
+    """
+
+    if request.method == "POST":
+        form = MyUserCreationForm(request.POST, request.FILES)
+
+        if form.is_valid():
+            user = form.save(commit=False)
+            user.save()
+
+            login(request, user)
+
+            return redirect("home")
+
+        else:
+            for errorKey in form.errors.keys():
+                value = form.errors[errorKey].get_context()["errors"][0]
+
+                if errorKey == "__all__":
+                    messages.error(request, f"{value}")
+                else:
+                    messages.error(request, f"{errorKey}: {value}")
+
+    form = MyUserCreationForm()
+
+    return render(request, "login/register_login.html", {'form': form, 'page': 'register'})
+
+
+@login_required(login_url="login")
+def user_logout(request):
+    """ user logout """
+
+    logout(request)
+    return redirect('login')
+
+
+@login_required(login_url="login")
+def main_page(request):
+    calandar = get_calandar_days()
+    current_month = get_current_month_name()
+    current_date = (datetime.now().month, datetime.now().day)
+    week_days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+
+    return render(
+        request, "user_site/home.html",
+        {'week_days': week_days, 'calandar': calandar, 'month_name': current_month, 'current_date': current_date}
+    )
+
+
+def get_current_month_name(month=None):
+    """ get the current month name.
+
+        if month is given then get the name of that month
+        otherwise get this month name
+    """
+    current_date = datetime.now()
+
+    if month is not None:
+        current_date = datetime(current_date.year, month, current_date.day)
+
+    return current_date.strftime("%B")
+
+
+def get_calandar_days(year=None, month=None):
+    """ return a matrix of dates. Each sublist is a week in the month, each week has 7 dates.
+        This list will include dates from the month before if the current month hasn't started with sunday """
+
+    current_week = list()
+    calendar = Calendar(6)  # 6 => means that the week start with Sunday
```

Comment:
In my opinion this is the only time we are going to use the calendar library,
 so I don't think overriding is necessary in this case.

Reply:
The idea was to try to prevent using the same variable name as an existing package/command
for example with list:
In [1]: l = list()

In [2]: l
Out[2]: []

In [3]: list = [1, 2, 3]

In [4]: list
Out[4]: [1, 2, 3]

In [5]: l2 = list()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-d0c38672fb70> in <module>
----> 1 l2 = list()

TypeError: 'list' object is not callable

Reply:
done